kclvm/sema/src/ty/context.rs

```diff
@@ -52,6 +55,27 @@ impl TypeContext {
         is_cyclic_directed(&self.dep_graph)
     }
 
+    /// Add dependencies and return the dependency graph is cyclic.
+    pub fn add_dependencies_and_is_cyclic(&mut self, from: &str, to: &str) -> bool {
```

Comment:
split `add_dependencies_and_is_cyclic` to `add_dependencies` and `self.is_cyclic()`